src/puppetlabs/puppetdb/command.clj

```diff
@@ -361,38 +363,51 @@
 
   (start [this context]
     (let [{:keys [scf-write-db catalog-hash-debug-dir]} (shared-globals)
-          config {:db scf-write-db
+          db-cfg {:db scf-write-db
                   :catalog-hash-debug-dir catalog-hash-debug-dir}
-          {:keys [response-chan response-pub]} context]
+          {:keys [response-chan response-pub]} context
+          factory (-> (conf/mq-broker-url (get-config))
+                      (mq/activemq-connection-factory))
+          connection (.createConnection factory)]
       (register-listener
        supported-command?
-       #(process-command-and-respond! % config response-chan (:stats context)))
-      context))
+       #(process-command-and-respond! % db-cfg response-chan (:stats context)))
+      (assoc context
+             :factory factory
+             :connection connection)))
 
   (stop [this context]
     (async/unsub-all (:response-pub context))
     (async/untap-all (:response-mult context))
     (async/close! (:response-chan-for-pub context))
     (async/close! (:response-chan context))
-    (dissoc context :response-pub :response-chan :response-chan-for-pub :response-mult))
+    (dissoc context :response-pub :response-chan :response-chan-for-pub :response-mult)
+    (.close (:connection context))
+    (.close (:factory context))
+    context)
 
   (stats [this]
     @(:stats (service-context this)))
 
-  (enqueue-command [this connection endpoint command version payload]
-    (enqueue-command this connection endpoint command version payload nil))
+  (enqueue-command [this command version payload]
+    (enqueue-command this command version payload nil))
 
-  (enqueue-command [this connection endpoint command version payload uuid]
-    (let [result (do-enqueue-command connection endpoint
+  (enqueue-command [this command version payload uuid]
+    (let [config (get-config)
+          connection (:connection (service-context this))
+          endpoint (get-in config [:command-processing :mq :endpoint])
+          command (if (string? command) command (command-names command))
```

Comment:
We talked about this some, I don't think it needs to be fixed in this PR, but it would be good not to have the mismatch here.